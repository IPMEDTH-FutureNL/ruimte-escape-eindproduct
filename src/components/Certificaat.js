import React, { Component } from 'react'
import  certificaat  from '../img/certificaat/certificaat.png';
import "../css/certificaat.css"

export class Certificaat extends Component {
  constructor(props){
      super(props);
  }

  printCertificate = (elem) =>{
    let mywindow = window.open('', '', 'height=794,width=1123');
    mywindow.document.write(`<head><link rel="preconnect" href="https://fonts.googleapis.com">
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
    <link href="https://fonts.googleapis.com/css2?family=Andika:ital,wght@0,400;0,700;1,400;1,700&display=swap" rel="stylesheet"> </head>`)
    mywindow.document.write(`<body>
                              <style> 
                                .elem{rotate(180deg)} 
                                .certificaat__punten { color: black; font-size: 50px; position: absolute; top: 50vh; left: 45.5vw;   font-family: 'Andika', sans-serif;} 
                                .certificaat { width: 1123px; height: 794px; border-radius: 25px;} @media print{@page {size: landscape}}
                                .certificaat__playerOne{
                                  color: black;
                                  font-size: 45px;
                                  position: absolute;
                                  top: 76vh;
                                  left: 30vh;
                                  font-family: 'Andika', sans-serif;
                                }
                              
                                .certificaat__playerTwo{
                                  color: black;
                                  font-size: 45px;
                                  position: absolute;
                                  top: 76vh;
                                  left: 88vh;
                                  font-family: 'Andika', sans-serif;
                                }
                              </style> `)

    mywindow.document.write(document.getElementById(elem).outerHTML);
    mywindow.document.write('<body/>');
    mywindow.document.close(); 
    mywindow.focus(); 
    mywindow.print();
  }

  toStart = () => {
    window.location.href = ('/');
  }

  render() {
    return (
        <section className="certificaat__container">
            <div id="elem" className="elem">
                <div>
                    <p className="certificaat__punten">Punten: {localStorage.getItem('punten')}</p>
                    <p className="certificaat__playerOne">{localStorage.getItem('playerOne')}</p>
                    <p className="certificaat__playerTwo">{localStorage.getItem('playerTwo')}</p>
                    <img src={certificaat} className="certificaat"/>
                   
                </div>
            </div>
            <button onClick={() => this.printCertificate('elem')} className="button__print">Print certificaat</button>
            <button onClick={() => this.toStart()} className="button__start">Naar start</button>
        </section>
    )
  }
}

export default Certificaat;