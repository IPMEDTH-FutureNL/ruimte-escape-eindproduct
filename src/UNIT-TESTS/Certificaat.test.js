import { render , screen, fireEvent } from "@testing-library/react"
import {BrowserRouter as Router} from 'react-router-dom';
import Certificaat from "../components/Certificaat"

test("check if hints escape get rendered on start up", () => {
    const { asFragment } = render(
        <Router>
            <Certificaat></Certificaat>
        </Router>
    )

    screen.debug();
    expect(asFragment()).toMatchSnapshot();
})