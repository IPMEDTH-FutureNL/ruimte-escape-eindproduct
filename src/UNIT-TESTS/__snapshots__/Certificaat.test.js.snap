// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if hints escape get rendered on start up 1`] = `
<DocumentFragment>
  <section
    class="certificaat__container"
  >
    <div
      class="elem"
      id="elem"
    >
      <div>
        <p
          class="certificaat__punten"
        >
          Punten: 
        </p>
        <p
          class="certificaat__playerOne"
        />
        <p
          class="certificaat__playerTwo"
        />
        <img
          class="certificaat"
          src="certificaat.png"
        />
      </div>
    </div>
    <button
      class="button__print"
    >
      Print certificaat
    </button>
    <button
      class="button__start"
    >
      Naar start
    </button>
  </section>
</DocumentFragment>
`;
